packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/marker.dart

```diff
@@ -292,8 +294,8 @@ class Marker implements MapsObject<Marker> {
       position: positionParam ?? position,
       rotation: rotationParam ?? rotation,
       visible: visibleParam ?? visible,
-      zIndex: zIndexParam ?? zIndex,
-      zIndexInt: zIndexIntParam ?? zIndexInt,
+      zIndex: zIndexIntParam != null ? 0 : (zIndexParam ?? 0),
+      zIndexInt: zIndexParam != null ? 0 : (zIndexIntParam ?? zIndexInt),
```

Comment:
These are backed by the same property; we should just always pass only zIndex for as long as both exist.

Reply:
This is not resolved; the latest version always uses `zIndexInt`, not `zIndex`, and thus destroys information on Android.